### ISSUE TITLE
Sources repository of http://apt.pop-os.org/proprietary

### ISSUE THREAD

**Issue by wmutschl** [pop-os/pop#311]
**Distribution (run `cat /etc/os-release`):**
 NAME="Pop!_OS"
VERSION="18.04 LTS"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Pop!_OS 18.04 LTS"
VERSION_ID="18.04"
HOME_URL="https://system76.com/pop"
SUPPORT_URL="http://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic

 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
apt:
  Installed: 1.6.1
  Candidate: 1.6.1
  Version table:
 *** 1.6.1 500
        500 http://us.archive.ubuntu.com/ubuntu bionic/main amd64 Packages
        100 /var/lib/dpkg/status

pop-shop:
  Installed: 0.2.9~1526005515~18.04~18df10f
  Candidate: 0.2.9~1526005515~18.04~18df10f
  Version table:
 *** 0.2.9~1526005515~18.04~18df10f 1001
       1001 http://ppa.launchpad.net/system76/pop/ubuntu bionic/main amd64 Packages
        100 /var/lib/dpkg/status

**Issue/Bug Description:**
If I add the source code for the repository for http://apt.pop-os.org/proprietary, I get the following message running apt update:

Hit:1 http://ppa.launchpad.net/nextcloud-devs/client/ubuntu bionic InRelease
Hit:2 http://shop.softmaker.com/repo/apt wheezy InRelease
Hit:3 http://repository.spotify.com stable InRelease                                                                   
Hit:4 http://us.archive.ubuntu.com/ubuntu bionic-updates InRelease                                                     
Hit:5 http://ppa.launchpad.net/system76/pop/ubuntu bionic InRelease                                                    
Hit:6 https://download.virtualbox.org/virtualbox/debian bionic InRelease                                           
Hit:7 http://ppa.launchpad.net/unit193/encryption/ubuntu bionic InRelease                                              
Hit:8 http://us.archive.ubuntu.com/ubuntu bionic-security InRelease                           
Hit:9 http://apt.pop-os.org/proprietary bionic InRelease                 
Hit:10 http://us.archive.ubuntu.com/ubuntu bionic-backports InRelease    
Hit:11 http://us.archive.ubuntu.com/ubuntu bionic InRelease
Reading package lists... Done                     
Building dependency tree       
Reading state information... Done
All packages are up to date.
**W: Skipping acquire of configured file 'main/source/Sources' as repository 'http://apt.pop-os.org/proprietary bionic InRelease' does not seem to provide it (sources.list entry misspelt?)**

When I remove the source code repository everything is fine.

**Steps to reproduce (if you know):**
This has been the case from the beginning.

**Expected behavior:**
apt update without skipping

**Other Notes:**

**Comment by brs17** [comment]
@wmutschl Are you able to access [http://apt.pop-os.org/proprietary/](http://apt.pop-os.org/proprietary/) in a web browser? 

How exactly are you adding the repository?

**Comment by wmutschl** [comment]
Yes I am able to access this in firefox. I think I enabled (not added) the repository via the Pop-Shop for the first time. However, it does not matter, whether I do so in the Pop-Shop or in the /etc/apt/sources.list
By the way, here is mine:

> wmutschl@pop-os:~$ cat /etc/apt/sources.list
> # deb cdrom:[Pop_OS 18.04 _Bionic Beaver_ - Release amd64 (20180612)]/ bionic main restricted
> deb http://de.archive.ubuntu.com/ubuntu/ bionic-updates main restricted universe multiverse
> deb-src http://de.archive.ubuntu.com/ubuntu/ bionic-updates main restricted universe multiverse #Added by software-properties
> deb http://de.archive.ubuntu.com/ubuntu/ bionic-security main restricted universe multiverse
> deb-src http://de.archive.ubuntu.com/ubuntu/ bionic-security main restricted universe multiverse #Added by software-properties
> deb http://de.archive.ubuntu.com/ubuntu/ bionic-backports main restricted universe multiverse
> deb-src http://de.archive.ubuntu.com/ubuntu/ bionic-backports main restricted universe multiverse #Added by software-properties
> deb http://de.archive.ubuntu.com/ubuntu/ bionic main restricted universe multiverse
> deb-src http://de.archive.ubuntu.com/ubuntu/ bionic main restricted universe multiverse #Added by software-properties
> deb http://apt.pop-os.org/proprietary bionic main
> deb-src http://apt.pop-os.org/proprietary bionic main

It's 
> deb-src http://apt.pop-os.org/proprietary bionic main
that causes the error with apt update

> All packages are up to date.
> W: Skipping acquire of configured file 'main/source/Sources' as repository 'http://apt.pop-os.org/proprietary bionic InRelease' does not seem to provide it (sources.list entry misspelt?)
> 

If I uncomment it, apt update runs through without warning.
In any case, I do not really need the source packages, but was just wondering about this issue.

Cheers,
Willi

**Comment by graphis** [comment]
I have exactly the same problems.
I disabled http://apt.pop-os.org/proprietary bionic main now, but I would like to know what the problem is

**Comment by mmstick** [comment]
@wmutschl There are no source packages in this repo, hence the error

```
deb-src http://apt.pop-os.org/proprietary bionic main
```

This entry shouldn't be in your sources list.

**Comment by wmutschl** [comment]
@mmstick that's what I was thinking. However, it is put there by default when installing popos (however it is uncommented by default)
yesterday I did a clean install of popos on my work Dell precision 7520 and there it was the same. But uncommenting it or deleting this line, solves this issue.
Maybe the devs should remove that from the default sources.list file?!

**Comment by mmstick** [comment]
> Maybe the devs should remove that from the default sources.list file?!

It shouldn't be getting added, as the ISO build script is configured to add the proprietary repo without the `--enable-source` argument to `add-apt-repository`.

**Comment by wmutschl** [comment]
Okay, but why is it still in the sources.list, if there are no source packages?

**Comment by mmstick** [comment]
This should be fixed now, since the Sources archive is generated now (even if it contains nothing).
